src/test/lib/ApiTest.cpp

```diff
@@ -1735,6 +1735,392 @@ class SecConfigTestContext {
     }
 };
 
+void QuicTestGlobalSetParam()
+{
+    //
+    // QUIC_PARAM_GLOBAL_SUPPORTED_VERSIONS
+    //
+    {
+        TestScopeLogger logScope("QUIC_PARAM_GLOBAL_SUPPORTED_VERSIONS is get only");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                nullptr,
+                QUIC_PARAM_GLOBAL_SUPPORTED_VERSIONS,
+                0,
+                nullptr));
+    }
+
+    //
+    // QUIC_PARAM_GLOBAL_PERF_COUNTERS
+    //
+    {
+        TestScopeLogger logScope("QUIC_PARAM_GLOBAL_PERF_COUNTERS is get only");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                nullptr,
+                QUIC_PARAM_GLOBAL_PERF_COUNTERS,
+                0,
+                nullptr));
+    }
+
+    //
+    // QUIC_PARAM_GLOBAL_LIBRARY_VERSION
+    //
+    {
+        TestScopeLogger logScope("QUIC_PARAM_GLOBAL_LIBRARY_VERSION is get only");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                nullptr,
+                QUIC_PARAM_GLOBAL_LIBRARY_VERSION,
+                0,
+                nullptr));
+    }
+
+    //
+    // QUIC_PARAM_GLOBAL_VERSION_SETTINGS
+    //
+    {
+        TestScopeLogger logScope("QUIC_PARAM_GLOBAL_VERSION_SETTINGS is covered by QuicTestVersionSettings");
+    }
+
+    //
+    // QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH
+    //
+    {
+        TestScopeLogger logScope("QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH is get only");
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                nullptr,
+                QUIC_PARAM_GLOBAL_LIBRARY_VERSION,
+                0,
+                nullptr));
+    }
+}
+
+void QuicTestCommonSetParam()
```

Comment:
Why isn't this separate? It looks like this test could go in the configuration one.

Reply:
The Parameter can by anything, but I could not find better param for "Anything"...
Is "0" ok?

Reply:
I am not quite sure what the goal is for this test then. Can you explain it a bit more please?

Reply:
This is testing here, beginning of SetParam

https://github.com/microsoft/msquic/blob/3b5d5cca496af1d5f80abc86ae0806bd0f417afd/src/core/api.c#L1302-L1320

Reply:
Removed to merge minimal test sets